attributes/default.rb

```diff
@@ -2,6 +2,9 @@
 default['packagecloud']['gpg_key_path'] = "/gpg.key"
 default['packagecloud']['hostname'] = nil
 
+default['packagecloud']['_stored_hostname'] = nil
+default['packagecloud']['override_hostname'] = nil
```

Comment:
@mfischer-zd we discussed this a bunch internally yesterday and this morning and I'm working on a separate PR right now (actually just re-running specs) that I think will address your concerns and the usability issues. I'm hoping to push that up shortly for review.